File: frontend/src/interfaces/document.ts
export interface DocumentInterface {
  id?: number;
  file_name?: string;
  file_type?: string;
  file_path?: string;
  upload_date?: string;  // ใน TypeScript ใช้ string สำหรับ date/time
  detail?: string;
  poster?: string;
  
  activity_id?: number;
  
  // ถ้าต้องการ relation กับ Activity (optional)
  activity?: {
    id: number;
    name: string;
    // fields อื่นๆ ของ Activity ที่ต้องการ
  };
  
  // จาก gorm.Model
  created_at?: string;
  updated_at?: string;
  deleted_at?: string | null;
}

File: backend/entity/document.go
package entity

import (
	"gorm.io/gorm"
	"time"
)

type Document struct {
	gorm.Model
	FileName 	string   `json:"file_name"`
	FileType    string    `json:"file_type"`
    FilePath    string    `json:"file_path"`
    UploadDate  time.Time `json:"upload_date"`
    Detail      string    `json:"detail"`
    Poster      string    `json:"poster"`

	LocationID *uint

	ProposalActivityID *uint              `json:"proposal_activity_id"`
	Activity           *Proposal_Activity `gorm:"foreignKey:ProposalActivityID"`
}